Arcane/Assets/Scripts/SceneScripts/GameScene/Piece/PieceScript/Magician.cs

```diff
@@ -1,20 +1,32 @@
 using UnityEngine;
 
-public class Magician : PieceObject
+public class Magician : ActivePieceObject
 {
+    public int counter = 1;
+    public override void ActiveEffect()
+    {
+        if (counter != 1) return;
+        GameManager.singleton.phaseMachine.TransitionTo(new MagicianPhase(this));
+    }
+
+    public override bool CanSpellActiveEffect()
+    {
+        return canActive;
```

Comment:
counterが0の時は、技が発動できないため、パネルが出てこなくて良さそう。

```suggestion
        if(counter == 0 ) return false;
        return canActive;
```